lazy_bench.py

```diff
@@ -0,0 +1,723 @@
+        return model(**inputs)
+    elif isistance(inputs, torch.Tensor):
+        return model(inputs)
+    raise RuntimeError("invalid example inputs ", inputs)
+
+class CudaSync:
+    def __init__(self, sync_every_iter=False):
+        self.sync_every_iter = sync_every_iter
+
+    def iter_sync(self):
+        if self.sync_every_iter:
+            torch.cuda.synchronize()
+
+    def final_sync(self):
+        torch.cuda.synchronize()
+
+class NoOpSync:
+    def __init__(self, sync_every_iter=False):
+        pass
+
+    def iter_sync(self):
+        pass
+
+    def final_sync(self):
+        pass
+
+class LazySync:
+    def __init__(self, sync_every_iter=False, skip_final_sync=False):
+        self.sync_every_iter = sync_every_iter
+        self.skip_final_sync = skip_final_sync
+
+    def iter_sync(self):
+        torch._lazy.mark_step()
+        if self.sync_every_iter:
+            torch._lazy.wait_device_ops()
+            if current_device == 'cuda':
+                torch.cuda.synchronize()
+
+    def final_sync(self):
+        torch._lazy.mark_step()
+        if self.skip_final_sync:
+            return
+        torch._lazy.wait_device_ops()
+        if current_device == 'cuda':
+            torch.cuda.synchronize()
+
+def dump_lazy_metrics(reset=False):
+    met = {name: int(metrics.counter_value(name)) for name in metrics.counter_names() if int(metrics.counter_value(name) > 0)}
+    if reset:
+        metrics.reset()
+    return met
+
+def timed(args, benchmark, sync, times=1):
+    results = None
+    sync.final_sync()
+    set_seeds()
+    if args.test == 'eval':
+        model, example_inputs = benchmark.get_module()
+
+    if current_device == 'lazy':
+        torch.cuda.set_sync_debug_mode(2)
+    elif current_device == 'cuda':
+        torch.cuda.set_sync_debug_mode(0)
+
+    # keep the lazy tensor results alive until the final sync
+    t0 = time.perf_counter()
+    for i in range(times):
+        if args.test == 'eval':
+            results = call_model_with(model, example_inputs)
+        elif args.test == 'train':
+            benchmark.train(niter=1)
+
+        # for the last i, let final_sync take care of it
+        if i < times - 1:
+            # may be just an async 'mark_step' for lazy, or no-op for cuda
+            sync.iter_sync()
+
+    if current_device in ['lazy', 'cuda']:
+        # don't assume torch.cuda present unless using cuda
+        torch.cuda.set_sync_debug_mode(0)
+
+    # should be a hard sync for lazy and cuda
+    # unless strictly measuring lazy trace overhead, then no-op
+    sync.final_sync()
+    t1 = time.perf_counter()
+    return results, t1 - t0
+
+def to_device(tensors, device):
+    """Handles moving tensor or tensors (in various containers) to a new device.
+        Used for various purposes (either correctness checking, or even as an impromptu
+        means of synchronization.) Note: this method doesn't apply a cuda sync, do that outside.
+    """
+
+    try:
```

Comment:
you know, at some point i hope we can delete all this. @xuzhao9 made some changes or at least proposed to make the output of eval() always tensors or tuples of tensors.  If that is done already, we might be able to just delete this.  But i think we should still land it as-is first.  one thing at a time..